packages/cli/README-source.md

```diff
@@ -1081,6 +1081,48 @@ This is an object that contains the data you returned from the `performSubscribe
+
+> The `bulkbundle` object is passed into the `performBulk` function as the second argument - IE: `performBulk: (z, bulkbundle) => {}`.
+
+### `bulkbundle.authData`
+
+It is a user-provided authentication data, like `api_key` or `access_token`. [Read more on authentication.](#authentication)
+
+### `bulkbundle.groupedBy`
+
+It is a user-provided data for a set of selected [`inputFields`](#input-fields) to group the multiple runs of a `create` action by.
+
+### `bulkbundle.bulk`
+
+It is a list of objects of user-provided data and some meta data to allow multiple runs of a `create` action be processed in a single API request.
+
+#### `bulkbundle.bulk[].inputData`
+
+It is a user-provided data for a particular run of a `create` action in the bulk, as defined by the [`inputFields`](#input-fields).
+
+#### `bulkbundle.bulk[].meta`
+
+It contains an idempotency `id` provided to the `create` action to identify each run's data in the bulk data.
+
+## Bulk-Write in Create Actions
+
+*Added in v15.8.0.*
+
+Bulk-write enables making a single API request for multiple Zap invocation runs on data having some specific data in common tagged as [`groupedBy`](https://github.com/zapier/zapier-platform/blob/main/packages/cli/README.md#bulkbundlegroupedby) in the `bulkbundle` object. It can ONLY be used in a `create` action. Within the `operation` object of the `create` action, a [`bulk`](https://github.com/zapier/zapier-platform/blob/main/packages/schema/docs/build/schema.md#bulkobjectschema) configuration object must be set along with a `performBulk` function to replace the general `perform` function. Note that the `perform` function cannot be defined along with the `performBulk` function. Check out the [`create` action operation schema](https://github.com/zapier/zapier-platform/blob/main/packages/schema/docs/build/schema.md#basiccreateactionoperationschema) for the details.
+
+Similar to the general `perform` function accepting 2 arguments [`z`](https://github.com/zapier/zapier-platform/blob/main/packages/cli/README.md#z-object) and [`bundle`](https://github.com/zapier/zapier-platform/blob/main/packages/cli/README.md#bundle-object) objects, the `performBulk` function accepts [`z`](https://github.com/zapier/zapier-platform/blob/main/packages/cli/README.md#z-object) and [`bulkbundle`](https://github.com/zapier/zapier-platform/blob/main/packages/cli/README.md#bulkbundle-object) objects. They share the same `z` object, but the `bulkbundle` object is different from the `bundle` object. The `bulkbundle` object has an idempotency ID set at `bulkbundle.bulk[].meta.id` for each invocation run in the bulk that would be used within the `performBulk` to tell Zapier backend which data items were successfully written. Find the details about the `bulkbundle` object [here](https://github.com/zapier/zapier-platform/blob/main/packages/cli/README.md#bulkbundle-object).
+
+Here is an example of a bulk-write usage in a `create` action:
```

Comment:
Suggested revision:

```markdown
## Bulk Create Actions

*Added in v15.8.0.*

Bulk Create allows you to create objects in bulk with a single or fewer API request(s). This is useful when you get want to reduce the number of requests made to your server. When enabled, Zapier holds the data until the buffer reaches a size limit or a certain time has passed, then sends the buffered data using the `performBulk` function you define.

To implement a Bulk Create, you define a [`bulk`](https://github.com/zapier/zapier-platform/blob/main/packages/schema/docs/build/schema.md#bulkobjectschema) configuration object and a `performBulk` function in the `operation` object. In the `bulk` config object, you specify how you want to group the buffered data using the `groupedBy` setting and the maximum number of items to buffer using the `limit` setting.

The `performBulk` function should replace the `perform` function. Note that `perform` cannot be defined along with `performBulk`. Check out the [`create` action operation schema](https://github.com/zapier/zapier-platform/blob/main/packages/schema/docs/build/schema.md#basiccreateactionoperationschema) for details.

Similar to the general `perform` function accepting two arguments, [`z`](https://github.com/zapier/zapier-platform/blob/main/packages/cli/README.md#z-object) and [`bundle`](https://github.com/zapier/zapier-platform/blob/main/packages/cli/README.md#bundle-object) objects, the `performBulk` function accepts [`z`](https://github.com/zapier/zapier-platform/blob/main/packages/cli/README.md#z-object) and [`bulkBundle`](https://github.com/zapier/zapier-platform/blob/main/packages/cli/README.md#bulkbundle-object) objects. They share the same `z` object, but the `bulkBundle` object is different from the `bundle` object. The `bulkBundle` object has an idempotency ID set at `bulkBundle.bulk[].meta.id` for each object in the bulk. `performBulk` would have to return the idempotency IDs to tell Zapier which objects were successfully written. Find the details about the `bulkBundle` object [here](https://github.com/zapier/zapier-platform/blob/main/packages/cli/README.md#bulkbundle-object).

Here is an example of a Bulk Create action:
```

Why:

* In most public-facing docs, we say "Create instead of "Write", so I'd suggest we say "Bulk Create" instead of "Bulk-Write Create".
* In the beginning, I started with a high-level description on why you would want to use a Bulk Create (to reduce the number of requests made to the partner's server).
* There isn't a notion of an "invocation run" in the current docs. It may be confusing for partners, so I removed it.

Let me know what you think. Thanks!

Reply:
Perfect, thanks.

Reply:
Updated